tests/System.Text.Primitives.Tests/EncodingPerfComparisonTests.cs

```diff
@@ -74,6 +135,59 @@ public string GenerateString(int charLength, int minCodePoint, int maxCodePoint)
             }
             return plainText.ToString();
         }
+
+        [Benchmark(InnerIterationCount = 250)]
+        [InlineData(1000, 0x0000, 0x007F)]
+        [InlineData(1000, 0x0080, 0x07FF)]
+        [InlineData(1000, 0x0800, 0xFFFF)]
+        public void DecodingPerformanceTestUsingCoreCLR(int charLength, int minCodePoint, int maxCodePoint)
+        {
+            string unicodeString = GenerateString(charLength, minCodePoint, maxCodePoint);
+            var bytes = Encoding.UTF8.GetBytes(unicodeString);
+
+            var decoder = Encoding.UTF8;
+            Assert.Equal(unicodeString, decoder.GetString(bytes));
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i=0; i<Benchmark.InnerIterationCount; i++)
+                    {
+                        decoder.GetString(bytes);
```

Comment:
@mgravell Can you change this to call the non-allocating version in the CLR (GetChars with pre-allocated buffers)?